packages/mui-material/src/SvgIcon/SvgIcon.js

```diff
@@ -85,28 +85,72 @@ const SvgIcon = React.forwardRef(function SvgIcon(inProps, ref) {
   };
 
   const more = {};
+  let newComponent;
+  let hasTitleAccess = titleAccess !== undefined;
 
   if (!inheritViewBox) {
     more.viewBox = viewBox;
   }
 
   const classes = useUtilityClasses(ownerState);
 
+  if (typeof component === 'function') {
+    const element = component({});
+
+    let mergedChildren = [].concat(element.props.children).concat(children);
+    if (titleAccess) {
+      mergedChildren.push(<title>{titleAccess}</title>);
+      hasTitleAccess = true;
+    }
+    if (element.props.titleAccess && !titleAccess) {
+      mergedChildren.push(<title>{element.props.titleAccess}</title>);
+      hasTitleAccess = true;
+    }
+
+    // Component can have no children, so remove undefined items.
+    // Give each object child a key.
+    mergedChildren = mergedChildren
+      .filter((child) => child !== undefined)
+      .map((child, index) => {
+        if (child instanceof Object) {
+          return {
+            ...child,
+            key: index,
+          };
+        }
+        return child;
+      });
+
+    // We have placed title, children and component children into mergedChildren array.
+    // If element has titleAccess prop, omit it to avoid creating duplicated title element.
+    // element.type can be a object or string tag.
+    newComponent = (componentProps) =>
+      element.props.titleAccess ? (
+        <element.type {...componentProps} {...element.props} {...more} titleAccess={undefined}>
+          {mergedChildren}
+        </element.type>
+      ) : (
+        <element.type {...componentProps} {...element.props} {...more}>
+          {mergedChildren}
+        </element.type>
+      );
+  }
```

Comment:
a new component should not be created at the render phase.

Reply:
I agree but I can't find out a better method if we need to use component prop as a function Component when the Component can use the children(https://github.com/mui/material-ui/blob/master/packages/mui-material/src/SvgIcon/SvgIcon.js#L108-L109).
You can close this PR if it isn't good or doesn't run in an MUI expected behavior.